WIP Initial attempt at adding Criu support to Docker

boucher opened the pull request:
This branch represents a merge of current Docker master with the libcontainer criu and swrk branches, along with @SaiedKazemi's original Docker criu branch. In addition to trying to get those branches working, a few features have been added tot he Criu support (in various states of completion), chiefly: being able to specify output directories and being able to leave the process running after checkpointing.

(removed committed vendor code from other branch)

cpuguy83 committed:
Cleanup daemon/volumes

- Mount struct now called volumeMount
- Merged volume creation for each volume type (volumes-from, binds, normal
  volumes) so this only happens in once place
- Simplified container copy of volumes (for when `docker cp` is a
  volume)

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

jessfraz committed:
Add docs for `--exec-opt` and setting `native.cgroupdriver`.

update man pages.
update bash completion.

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle <jess@docker.com> (github: jfrazelle)

Docker-DCO-1.1-Signed-off-by: Jessie Frazelle <jess@docker.com> (github: jfrazelle)

cyphar committed:
*: expose getResourcePath and getRootResourcePath wrappers

Due to the importance of path safety, the internal sanitisation wrappers
for volumes and containers should be exposed so other parts of Docker
can benefit from proper path sanitisation.

Signed-off-by: Aleksa Sarai <cyphar@cyphar.com> (github: cyphar)

jessfraz committed:
add initial docs for windows client testing

Docker-DCO-1.1-Signed-off-by: Jessie Frazelle <princess@docker.com> (github: jfrazelle)

Docker-DCO-1.1-Signed-off-by: Jessie Frazelle <jess@docker.com> (github: jfrazelle)

docs: crop and optimize images

Cropped and optimized images.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Docker-DCO-1.1-Signed-off-by: Jessie Frazelle <princess@docker.com> (github: jfrazelle)

Testing and updating the Windows material

Signed-off-by: Mary Anthony <mary@docker.com>

checkpoint

Signed-off-by: Mary Anthony <mary@docker.com>

Fitting the Windows specific material into the contributor guide

Signed-off-by: Mary Anthony <mary@docker.com>

Entering James comments

Signed-off-by: Mary Anthony <mary@docker.com>

cyphar committed:
*: switch to Get(Root)?ResourcePath where appropriate

Several parts of the codebase didn't use the correct path sanitisation
wrappers. Now that the wrappers have been exposed, use those.

Signed-off-by: Aleksa Sarai <cyphar@cyphar.com> (github: cyphar)

mizzy committed:
Append icc related iptables rules, not INSERT

Signed-off-by: Gosuke Miyashita <gosukenator@gmail.com>

brahmaroutu committed:
Allow go template to work properly with inspect
Closes #11641

Signed-off-by: Srini Brahmaroutu <srbrahma@us.ibm.com>

jessfraz committed:
option to configure cgroups

Signed-off-by: Jessica Frazelle <jess@docker.com>

buddhamagnet committed:
add support for exclusion rules in dockerignore

Signed-off-by: Dave Goodchild <buddhamagnet@gmail.com>

boucher committed:
Initial attempt at adding Criu support to Docker

This branch represents a merge of current Docker master with the libcontainer criu and swrk branches, along with @SaiedKazemi's original Docker criu branch. In addition to trying to get those branches working, a few features have been added tot he Criu support (in various states of completion), chiefly: being able to specify output directories and being able to leave the process running after checkpointing.

SaiedKazemi commented:
@boucher 
This is great.  Thanks for integrating C/R into the latest Docker head.  The review process on my part will unfortunately be slow due to other high priority activities and then traveling (vacation/work).

Commit:
Remove job from container_inspect

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

Commit:
clean up viz code

Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

hqhq committed:
add devices cgroup check and errors

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq committed:
simplify memory limit check

If memory cgroup is mounted, memory limit is always supported,
no need to check if these files are exist.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

SvenDowideit committed:
Copy over the DHE documentation for release to docs.docker.com

Signed-off-by: Sven Dowideit <SvenDowideit@home.org.au>

Commit:
Putting into our new format for cloud
Adding in Seb's comments
Updating with Fred's comments

Signed-off-by: Mary Anthony <mary@docker.com>

wonderflow committed:
remove execCreate & execStart from job

Also removed the function ExecConfigFromJob

Signed-off-by: Sun Jianbo <wonderflow@zju.edu.cn>
Signed-off-by: Alexander Morozov <lk4d4@docker.com>

tianon committed:
Fix daemon start/stop logic in hack/make/* scripts

From the Bash manual's `set -e` description:
(https://www.gnu.org/software/bash/manual/bashref.html#index-set)

> Exit immediately if a pipeline (see Pipelines), which may consist of a
> single simple command (see Simple Commands), a list (see Lists), or a
> compound command (see Compound Commands) returns a non-zero status.
> The shell does not exit if the command that fails is part of the
> command list immediately following a while or until keyword, part of
> the test in an if statement, part of any command executed in a && or
> || list except the command following the final && or ||, any command
> in a pipeline but the last, or if the command’s return status is being
> inverted with !. If a compound command other than a subshell returns a
> non-zero status because a command failed while -e was being ignored,
> the shell does not exit.

Additionally, further down:

> If a compound command or shell function executes in a context where -e
> is being ignored, none of the commands executed within the compound
> command or function body will be affected by the -e setting, even if
> -e is set and a command returns a failure status. If a compound
> command or shell function sets -e while executing in a context where
> -e is ignored, that setting will not have any effect until the
> compound command or the command containing the function call
> completes.

Thus, the only way to have our `.integration-daemon-stop` script
actually run appropriately to clean up our daemon on test/script failure
is to use `trap ... EXIT`, which we traditionally avoid because it does
not have any stacking capabilities, but in this case is a reasonable
compromise because it's going to be the only script using it (for now,
at least; we can evaluate more complex solutions in the future if they
actually become necessary).

The alternatives were much less reasonable.  One is to have the entire
complex chains in any script wanting to use `.integration-daemon-start`
/ `.integration-daemon-stop` be chained together with `&&` in an `if`
block, which is untenable.  The other I could think of was taking the
body of these scripts out into separate scripts, essentially meaning
we'd need two files for each of these, which further complicates the
maintenance.

Add to that the fact that our `trap ... EXIT` is scoped to the enclosing
subshell (`( ... )`) and we're in even more reasonable territory with
this pattern.

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

calavera committed:
Shallow clone using git to build images.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
Fix TestEventsImageImport racy, fixes #12499

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

LK4D4 committed:
Add gocheck to vendored deps

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

LK4D4 committed:
Use suite for integration-cli

It prints test name and duration for each test.
Also performs deleteAllContainers after each test.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

SvenDowideit committed:
make space for DHE menu item again

Signed-off-by: Sven Dowideit <SvenDowideit@home.org.au>

rhatdan committed:
Add journald as a supported logger for containers

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

ankushagarwal committed:
Eliminate json.Marshal from graph/export.go and volumes/volume.go

Fixes #12531
Fixes #12530

Signed-off-by: Ankush Agarwal <ankushagarwal11@gmail.com>

kostickm committed:
Fixing statusCode checks for sockRequest

Signed-off-by: Megan Kostick <mkostick@us.ibm.com>

ahmetb committed:
docs: Add more places docker.service can be at

More systemd goodness. Documenting where `docker.service` lives under Ubuntu 15.

Signed-off-by: Ahmet Alp Balkan <ahmetalpbalkan@gmail.com>

Commit:
Move https integration tests as unit tests under client
Addresses #12255
Signed-off-by: Srini Brahmaroutu <srbrahma@us.ibm.com>

jessfraz committed:
actually depreciate -rm, -sig-proxy, -name, seriously its been forever

Signed-off-by: Jessica Frazelle <princess@docker.com>

Commit:
Merge pull request #12539 from WeiZhang555/httpError

change httpError logic

Commit:
Merge pull request #12517 from mavenugo/master

Enhanced port integration-cli tests

Commit:
Merge pull request #12457 from calavera/remove_engine_job_builder

Remove engine.Job from Builder.

Commit:
Merge pull request #12430 from icecrime/11551_carry

Add -H|--human command-line flag to `docker history`

tianon committed:
Merge pull request #12181 from inatatsu/enable-netgo-with-gccgo

Enable "netgo" library when we build with gccgo

Commit:
Merge pull request #10736 from coolljt0725/add_cpu_limit

Add support cpu cfs_quota

dalanlan committed:
remove job from load
Signed-off-by: Simei He <hesimei@zju.edu.cn>

Signed-off-by: He Simei <hesimei@zju.edu.cn>

dalanlan committed:
remove job from push
Signed-off-by: Simei He <hesimei@zju.edu.cn>

Signed-off-by: He Simei <hesimei@zju.edu.cn>

ankushagarwal committed:
Add integration test for history option

Parse the history output to locate the size fields and check
whether they are the correct format or not. Use Column name SIZE
to mark start and end indices of the size fields

Fixes #12578

Signed-off-by: Ankush Agarwal <ankushagarwal11@gmail.com>

dalanlan committed:
remove job from image_export

Signed-off-by: He Simei <hesimei@zju.edu.cn>
Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Makes headings in documentation consistent

Fixes #10673.

Signed-off-by: Rick Wieman <git@rickw.nl>

jamtur01 committed:
Merge pull request #12563 from kentoj/patch-1

Clarify data volume init with existing data

Commit:
Remove writeJSONEnv

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

LK4D4 committed:
Merge pull request #12598 from runcom/remove-writeJSONEnv

Remove writeJSONEnv, fixes #12597

LK4D4 committed:
Remove image_set engine job

It was unused

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Remove job image_get

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

calavera committed:
Do not try to shallow git history when the protocol doesn't allow it.

This only happens with the old git http dumb protocol, but that's what we use in our integration tests.
We check the Content-Type header advertised in http requests to make sure the http transport is the git smart transport:

See this commit as a reference:
https://github.com/git/git/commit/4656bf47fca857df51b5d6f4b7b052192b3b2317

Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
Remove job image_tarlayer

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

cpuguy83 committed:
Merge pull request #12604 from runcom/remove-job-image-tarlayer

Remove job image_tarlayer

duglin committed:
Merge pull request #12601 from runcom/remove-job-image-get

Remove job image_get

ankushagarwal committed:
Add Image Digest doc in userguide/dockerimages

Fixes #12551

Signed-off-by: Ankush Agarwal <ankushagarwal11@gmail.com>

Commit:
Remove not needed call to container.readHostConfig()

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

LK4D4 committed:
Merge pull request #12609 from runcom/remove-not-needed-readHostConfig

Remove not needed call to container.readHostConfig()

cpuguy83 committed:
Merge pull request #12564 from LK4D4/remove_jobs_trust

Remove engine from trust

cpuguy83 committed:
Merge pull request #12600 from LK4D4/remove_image_set

Remove image_set engine job

LK4D4 committed:
Merge pull request #12567 from brahmaroutu/integration_test6_12255

moving integration tests to graph unit tests

Commit:
Merge pull request #12536 from mountkin/remove-unsupported-instructions-from-doc

Remove some unsupported instructions in the docs.

LK4D4 committed:
Remove chain of engine passing from builder to loadManifest

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

calavera committed:
Test that we set the right arguments for git cloning depending on the transport.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Commit:
Merge pull request #12511 from LK4D4/gocheck_suites

Gocheck suites

Commit:
Replace json.Unmarshal with json.Decoder().Decode()

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

cpuguy83 committed:
Merge pull request #12611 from LK4D4/remove_eng_chain_pull

Remove chain of engine passing from builder to loadManifest

hqhq committed:
remove redundant warning

And warning is not supposed to have a prefix WARNING.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

ankushagarwal committed:
Add Configuring Docker article [WIP]

Fixes #12088

Signed-off-by: Ankush Agarwal <ankushagarwal11@gmail.com>

Commit:
Fixing a few links in registry

Signed-off-by: Mary Anthony <mary@docker.com>

cpuguy83 committed:
remove some uneeded sleeps in tests

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

psftw committed:
Rewrite Official Repositories page.

The existing page is focused on listing a set of requirements for
proposing a new repository.  This information has become outdated and is
duplicated in the `docker-library/official-images` and
`docker-library/docs` GitHub repositories.  This PR rewrites the
Official Repositories page to describe what they actually are, and
defers to GitHub/IRC for the subset of users that are interested in
contributing.  I also removed the requirement to contact
partners@docker.com and made it optional to reduce the barrier to entry.

Signed-off-by: Peter Salvatore <peter@psftw.com>

LK4D4 committed:
Wait until all pushes are done in TestPushInterrupt

Background pushes affects other tests

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

peggyl committed:
Fix golint errors in docker/api/client

Signed-off-by: Peggy Li <peggyli.224@gmail.com>

sublimino committed:
Add kali to install script

https://www.kali.org/ is a Debian derivative. This script completes
succesfully using the Debian install path

Signed-off-by: Andrew Martin <sublimino@gmail.com>

PeterDaveHello committed:
Use svg instead of png to get better image quality

Signed-off-by: Peter Dave Hello <hsu@peterdavehello.org>

rajdeepd committed:
TestCase added for Container Create with HostName

Signed-off-by: Rajdeep Dua <dua_rajdeep@yahoo.com>

Commit:
Move CLI config processing out from under registry dir

No logic changes should be in here, just moving things around.

Signed-off-by: Doug Davis <dug@us.ibm.com>

ankushagarwal committed:
Update with @moxiegirl's patch and add direct config

Signed-off-by: Ankush Agarwal <ankushagarwal11@gmail.com>

Commit:
Merge pull request #12634 from PeterDaveHello/patch-2

Use svg instead of png to get better image quality

tianon committed:
Merge pull request #12637 from sublimino/patch-1

Add kali to install script

cpuguy83 committed:
Merge pull request #12623 from peggyl/api-client-docs

Fix golint errors in docker/api/client

Commit:
Validate we're not using the old testing stuff

Signed-off-by: Doug Davis <dug@us.ibm.com>

Commit:
Remove old testing stuff that slipped into master

Signed-off-by: Doug Davis <dug@us.ibm.com>

cpuguy83 committed:
Merge pull request #12647 from duglin/FixTesting

Remove old testing stuff that slipped into master

jessfraz committed:
add integration test for error pull nonexistent

Signed-off-by: Jessica Frazelle <princess@docker.com>

Commit:
Merge pull request #12376 from Mic92/refactor-hack

Refactor shellscripts

Commit:
Correctly format API error on image pull

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

LK4D4 committed:
Merge pull request #12588 from hqhq/hq_fix_cpuquota_warn

remove redundant warning

jessfraz committed:
update contrib docs for gocheck

Signed-off-by: Jessica Frazelle <princess@docker.com>

LK4D4 committed:
Merge pull request #12621 from LK4D4/test_push_interrupt_daemon

Run TestPushInterrupt on separate daemon

jessfraz committed:
add regression test for rmi multiple tags without f

Signed-off-by: Jessica Frazelle <princess@docker.com>

rhatdan committed:
Merge branch 'master' of github.com:docker/docker into journald

Docker-DCO-1.1-Signed-off-by: Dan Walsh <dwalsh@redhat.com> (github: rhatdan)

Commit:
Validate repo name before image pull

Checks for reserved 'scratch' image name.

fixes #12281

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

dqminh committed:
remove unused utils

Signed-off-by: Daniel, Dao Quang Minh <dqminh89@gmail.com>

LK4D4 committed:
Merge pull request #12657 from dqminh/remove-unused-utils

remove unused utils

Commit:
Merge pull request #12650 from LK4D4/carry_12472

remove execCreate & execStart from job

SvenDowideit committed:
Merge pull request #12460 from moxiegirl/fix-carry-12383

Putting into our new format for cloud

SvenDowideit committed:
Merge pull request #12513 from allingeek/patch-1

builder.md - Changes to bring LABEL docs in line with 1.6 release.

Commit:
Merge pull request #10992 from cpuguy83/add_volume_mounting_for_cp

Make `docker cp` bind-mount volumes

SvenDowideit committed:
Merge pull request #12586 from ankushagarwal/image-digests

Add Image Digest doc in userguide/dockerimages

dqminh committed:
reuse same code for setting pipes in run/exec

This also moves `exec -i` test to _unix_test.go because it seems to need a
pty to reliably reproduce the behavior.

Signed-off-by: Daniel, Dao Quang Minh <dqminh89@gmail.com>

SvenDowideit committed:
Merge pull request #12614 from ankushagarwal/ubuntu-docs

Add Configuring Docker article

SvenDowideit committed:
Merge pull request #12615 from moxiegirl/update-repository-links

Fixing a few links in registry

Commit:
Merge pull request #10422 from cpuguy83/cleanup_daemon_volumes

Cleanup daemon/volumes

LK4D4 committed:
Merge pull request #12557 from rhatdan/journald

Add journald as a supported logger for containers

duglin committed:
Merge pull request #12649 from jlhawn/fix_pull_err_explosion

Correctly format API error on image pull

Commit:
Merge pull request #12652 from jfrazelle/update-contrib-docs-for-go-check

update contrib docs for gocheck

cpuguy83 committed:
Merge pull request #12406 from runcom/remove-job-container-inspect

Remove job from container_inspect

cpuguy83 committed:
Merge pull request #12654 from jfrazelle/12135-docker-delete-multiple-tagged

add regression test for rmi multiple tags without f

hqhq committed:
fix inspect format result

Currently `docker inspect -f` use json.Unmarshal() unmarshal
to interface, it will store all JSON numbers in float64, so
we use `docker inspect 4f0d73b75a0d | grep Memory` and
`docker inspect -f {{.HostConfig.Memory}} 4f0d73b75a0d` will
get different values.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

cpuguy83 committed:
Merge pull request #12543 from vdemeester/11584-pkg-stdcopy-test-coverage

Add some stdcopy_test (coverage)

cpuguy83 committed:
Merge pull request #12582 from ankushagarwal/add-docker-history-integration-test

Add integration test for history option

Commit:
Move setHostConfig to daemon file

Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

duglin committed:
Merge pull request #12390 from hqhq/hq_fix_inspect

fix inspect format result

TomasTomecek committed:
v1 spec: fix typos and formatting

Signed-off-by: Tomas Tomecek <ttomecek@redhat.com>

SvenDowideit committed:
Merge pull request #12599 from RickWieman/10673-make-headings-consistent

Makes headings in documentation consistent

vdemeester committed:
Fix the design proposal link in advanced contribution page

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Commit:
push test: fix typo

Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

Commit:
Fix typo in builder reference

Signed-off-by: Tristan Carel <tristan.carel@gmail.com>

hqhq committed:
fix test case name

Name like this will never run by go test. And this test case won't
get PAAS.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

WeiZhang555 committed:
refactor httpError() and add 404 "not found" mapping

When docker pull a non-existent repo, daemon will report "image xxx not found"
with an error code 500, which should be 404.
This commit add 404 "not found" mapping and refactor httpError function.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

duglin committed:
Merge pull request #12680 from Mashimiao/fix-typo-push-test

push test: fix typo

duglin committed:
Merge pull request #12663 from Mashimiao/Move-setHostConfig-to-daemon-file

Move setHostConfig to daemon file

duglin committed:
Merge pull request #12579 from ZJU-SEL/remove_job_from_load

remove job from load

vbatts committed:
contrib/init: unshare mount namespace for inits

* openrc
* sysvinit-debian
* upstart

Signed-off-by: Vincent Batts <vbatts@redhat.com>

cpuguy83 committed:
Fix race with TestContainerApiCommit

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Fix TestRenameStoppedContainer race

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

Commit:
Remove deleteAllContainers call in test

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

cpuguy83 committed:
Merge pull request #12683 from tristan0x/patch-1

Fix typo in builder reference

LK4D4 committed:
Merge pull request #12681 from hqhq/hq_fix_test_in_apiserver

fix test case name

calavera committed:
Document the extra `depth` argument in git contexts.

Signed-off-by: David Calavera <david.calavera@gmail.com>

LK4D4 committed:
Merge pull request #12505 from ZJU-SEL/remove_job_from_push

remove job from push

LK4D4 committed:
Merge pull request #12691 from runcom/remove-deleteAllContainers-call

Remove deleteAllContainers call in test

calavera committed:
Remove duplicated git clone logic.

Signed-off-by: David Calavera <david.calavera@gmail.com>

LK4D4 committed:
Return weird behaviour of returning json errors

We need this, so client can get error from stream and not from status
code, which is already 200, because write to ResponseWriter was occured.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

LK4D4 committed:
Merge pull request #12500 from tianon/fix-daemon-logic

Fix daemon start/stop logic in hack/make/* scripts

jfrazelle committed:
Merge pull request #12687 from WeiZhang555/NotFound

refactor httpError() and add 404 "not found" mapping

LK4D4 committed:
Merge pull request #12646 from duglin/validateTests

Validate we're not using the old testing stuff

cpuguy83 committed:
Merge pull request #12703 from LK4D4/carry_12596

remove job from image_export

jfrazelle committed:
Merge pull request #12692 from cpuguy83/fix_commit_test

Fix race with TestContainerApiCommit

jfrazelle committed:
Merge pull request #12690 from rajdeepd/dry-run-test

Container API Test with HostName

DarianAnjuhal committed:
fix runtime issue for TestEventsFilterContainer

Signed-off-by: Daniel Antlinger <d.antlinger@gmx.at>
Signed-off-by: Alexander Morozov <lk4d4@docker.com>

LK4D4 committed:
Merge pull request #12636 from duglin/MoveConfig

Move CLI config processing out from under registry dir

Commit:
Merge pull request #12577 from jfrazelle/actually-depreciate-rm

actually depreciate -rm insteam of --rm.. and a few others

Commit:
Remove engine/job from graph

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

jfrazelle committed:
Merge pull request #12702 from LK4D4/good_ol_flushed

Return weird behaviour of returning json errors

jfrazelle committed:
Merge pull request #12693 from runcom/fix-TestRenameStoppedContainer-race

Fix TestRenameStoppedContainer race

jfrazelle committed:
Merge pull request #12674 from TomasTomecek/fix-v1-spec-typos

v1 spec: fix typos and formatting

LK4D4 committed:
Merge pull request #11839 from brahmaroutu/template_11641

Allow go template to work properly with inspect …

jfrazelle committed:
Merge pull request #12655 from jlhawn/fix_12281

Validate repo name before image pull

jfrazelle committed:
Merge pull request #12575 from brahmaroutu/integration_test7_12255

Move https tests as unit tests under client

vdemeester committed:
delete "defer deleteContainer" on tests

Since docker test suite is now using gocheck, ``defer
deleteContainer(…)`` is not needed anymore.

Fixes #12705

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

jfrazelle committed:
Merge pull request #12509 from runcom/12499-racy-TestEventsImageImport

Fix TestEventsImageImport racy

jfrazelle committed:
Merge pull request #12417 from hashneo/patch-1

What if authConfig or factory is Nill?

tiborvass committed:
Merge pull request #9397 from jpopelka/9395-firewalld

Firewalld support

Commit:
Removing firewalld info

Signed-off-by: Mary Anthony <mary@docker.com>

fw42 committed:
[docs] fix formatting issue

Signed-off-by: Florian Weingarten <flo@hackvalue.de>

LK4D4 committed:
Merge pull request #12709 from vdemeester/12705-remove-defer-deleteContainer

delete "defer deleteContainer" on tests

noxiouz committed:
statsCollector: fix data race in run()

statsCollector.publishers must be protected to prevent
modifications during the iteration in run().
Being locked for a long time is bad, so pairs of containers &
publishers (pointers) are copied to release the lock fast.

Signed-off-by: Anton Tiurin <noxiouz@yandex.ru>

LK4D4 committed:
Implement teardown removeAllImages

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

Commit:
Merge pull request #12711 from Shopify/fix_doc

[docs] small markup fix

jfrazelle committed:
Merge pull request #12710 from runcom/bye-engine-from-graph

Remove engine/job from graph

crosbymichael committed:
Merge pull request #12659 from dqminh/exec-interactive-hang

reuse same code for setting pipes in run/exec

jfrazelle committed:
Merge pull request #12569 from kostickm/12449-check-statusCode

Fixing statusCode checks for sockRequest

Commit:
Merge pull request #12712 from moxiegirl/carry-docs-update-9397

Removing firewalld info

coolljt0725 committed:
Add cpu cfs quota to build

Signed-off-by: Lei Jitang <leijitang@huawei.com>

cpuguy83 committed:
Merge pull request #12587 from coolljt0725/add_cpu_cfs_quota_to_build

Add cpu cfs quota to build

Commit:
Merge pull request #12729 from SvenDowideit/dhe-documentation-master-pr

Dhe documentation master pr

HuKeping committed:
Add test for REST API container rename

Signed-off-by: Hu Keping <hukeping@huawei.com>

vdemeester committed:
Add coverage on pkg/archive

Add tests on:
- changes.go
- archive.go
- wrap.go

Should fix #11603 as the coverage is now 81.2% on the ``pkg/archive``
package. There is still room for improvement though :).

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

estesp committed:
Merge pull request #12678 from vdemeester/fix-advanced-contributing-doc

Documentation: fix the design proposal link in advanced contribution page

LK4D4 committed:
Merge pull request #12502 from calavera/shallow_git_history

Shallow clone using git to build images.

LK4D4 committed:
Use -f for rm instead of kill before

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

duglin committed:
Merge pull request #12753 from LK4D4/remove_all_images

Implement teardown removeAllImages

cpuguy83 committed:
Merge pull request #12754 from LK4D4/carry_12707

fix runtime issue for TestEventsFilterContainer

ibuildthecloud committed:
Expose ParseRestartPolicy

ParseRestartPolicy is useful function for third party go programs to use
so that they can parse the restart policy in the same way that Docker
does

Signed-off-by: Darren Shepherd <darren@rancher.com>

cpuguy83 committed:
Merge pull request #12755 from LK4D4/use_f_for_remove

Use -f for rm instead of kill before

LK4D4 committed:
Not protect dangling images for integration-cli

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

cpuguy83 committed:
Merge pull request #12762 from ibuildthecloud/export-restart-parse

Expose ParseRestartPolicy

LK4D4 committed:
Implement DockerRegistrySuite in integration-cli

To avoid manually creating and destroying registrys in tests.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

LK4D4 committed:
Merge pull request #12764 from LK4D4/not_protect_dangling

Not protect dangling images for integration-cli

Commit:
Fixed typo 'configuring'

Signed-off-by: John Howard <jhoward@microsoft.com>

jfrazelle committed:
Merge pull request #12775 from Microsoft/12774-fixtypo

Fixed typo 'configuring'

jfrazelle committed:
Merge pull request #12699 from vbatts/vbatts-init_mnt_ns

contrib/init: unshare mount namespace for inits

estesp committed:
Merge pull request #12765 from LK4D4/registry_suite

Implement DockerRegistrySuite in integration-cli

LK4D4 committed:
Merge pull request #12744 from vdemeester/11603-pkg-archive-test-coverage-2

Add coverage on pkg/archive (related to #11603)

sergey-alekseev committed:
Fix documentation typo in articles/basics.md

Signed-off-by: Sergey Alekseev <sergey.alekseev.minsk@gmail.com>

Commit:
A fix for = in env values in linked containers

Closes: #12763

Signed-off-by: Doug Davis <dug@us.ibm.com>

Commit:
Fix race condition in API commit test

Signed-off-by: Doug Davis <dug@us.ibm.com>

LK4D4 committed:
Merge pull request #12784 from duglin/RaceCommit

Fix race condition in API commit test

epc committed:
Copy edits for typos

Signed-off-by: Ed Costello <epc@epcostello.com>

LK4D4 committed:
Merge pull request #12728 from HuKeping/addtest

Add test for REST API container rename

LK4D4 committed:
Implement daemon suite for integration-cli

For creating and stopping test daemons automatically.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

calavera committed:
Merge pull request #12734 from sergey-alekseev/sergey-alekseev-patch-1

Fix documentation typo in articles/basics.md

Commit:
Small if err cleaning

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

Commit:
Adding James and theJeztah to the list

Signed-off-by: Mary Anthony <mary@docker.com>

Commit:
Merge pull request #12787 from epc/copyedits

Copy edits for typos

dnephin committed:
Add missing API docs about filtering by label.

Signed-off-by: Daniel Nephin <dnephin@gmail.com>

aidanhs committed:
Prevent deadlock on attempt to use own net

Signed-off-by: Aidan Hobson Sayers <aidanhs@cantab.net>

thewtex committed:
Do Debian installation verification with hello-world image.

The hello-world image is recommended as a verification image and it is smaller
than Ubuntu.

Signed-off-by: Matt McCormick <matt.mccormick@kitware.com>

dalanlan committed:
remove useless http call from export
Signed-off-by: He Simei <hesimei@zju.edu.cn>

jmzwcn committed:
Remove dead code from daemon/daemon.go

fix #12492
Signed-off-by: Daniel Zhang <jmzwcn@gmail.com>

Commit:
Remove c.Fatal from goroutine in TestGetContainersAttachWebsocket

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

WeiZhang555 committed:
improve docker man page

fix issue #12708:
"docker man page is misleading about format of commands"

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

duglin committed:
Merge pull request #12616 from cpuguy83/remove_uneeded_sleeps

remove some uneeded sleeps in tests

cpuguy83 committed:
Merge pull request #12783 from duglin/Issue12763-LinkedEnvs

A fix for = in env values in linked containers

vbatts committed:
rhel.md: update RHEL6 from 6.5 to 6.6

Signed-off-by: Vincent Batts <vbatts@redhat.com>

vbatts committed:
rhel.md: bump the kernel version for RHEL6

Closes #9856

Signed-off-by: Vincent Batts <vbatts@redhat.com>

estesp committed:
Merge pull request #12803 from ZJU-SEL/remove_useless_export_httpcall

remove useless http call from export

Commit:
Windows: Info no containerized check

Signed-off-by: John Howard <jhoward@microsoft.com>

LK4D4 committed:
Merge pull request #12799 from aidanhs/aphs-fix-net-container-deadlock

Prevent deadlock on --net=container:<self>

jfrazelle committed:
Merge pull request #12714 from noxiouz/fix_race_stats_collector

statsCollector: fix data race in run()

LK4D4 committed:
Merge pull request #12809 from runcom/remove-cfatal-goroutine-attachws

Remove c.Fatal from goroutine in TestGetContainersAttachWebsocket

cpuguy83 committed:
Merge pull request #12807 from jmzwcn/patch-2

Remove dead code from daemon/daemon.go

Commit:
Windows: Refactor volumes

Signed-off-by: John Howard <jhoward@microsoft.com>

Commit:
Expose whole Response struct in sockRequestRaw

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

cpuguy83 committed:
Merge pull request #12818 from Microsoft/10662-nocontainercheckonwindows

Windows: Info no containerized check

tiborvass committed:
Merge pull request #12793 from runcom/bye-unmarshal

Replace json.Unmarshal() with json.Decoder().Decode()

calavera committed:
Merge pull request #12819 from Microsoft/10662-volumesrefactor

Windows: Refactor volumes

Commit:
c.Fatal won't fail and exit test inside a goroutine, errors should be handled outside with a channel

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

tiborvass committed:
Merge pull request #12333 from Mashimiao/clean-up-viz

Clean up viz code

vbatts committed:
rhel.md: adding link to most recent issue

Signed-off-by: Vincent Batts <vbatts@redhat.com>

jfrazelle committed:
Merge pull request #12532 from ankushagarwal/eliminate-json-marshal

Eliminate json.Marshal from graph/export.go and volumes/volume.go

fntlnz committed:
Parallelize TestEventsLimit

Signed-off-by: Lorenzo Fontana <fontanalorenzo@me.com>

cpuguy83 committed:
Merge pull request #12719 from fntlnz/12675-test-events-emit

Parallelize TestEventsLimit

cpuguy83 committed:
Fix undead containers

When a container has errors on removal, it gets flagged as dead.
If you `docker rm -f` a dead container the container is dereffed from
the daemon and doesn't show up on `docker ps` anymore... except that the
container JSON file may still be lingering around and becomes undead
when you restart the daemon.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

tdmackey committed:
trivial: typo cleanup

Signed-off-by: David Mackey <tdmackey@booleanhaiku.com>

tiborvass committed:
Merge pull request #12794 from runcom/small-cleaning

Small if err cleaning

estesp committed:
Merge pull request #12828 from tdmackey/trivial-spelling

trivial: typo cleanup

Commit:
Merge pull request #12801 from thewtex/doc-test-ubuntu

Do Debian installation verification with hello-world image.

cpuguy83 committed:
Merge pull request #11882 from hqhq/hq_warn_device_cg

add devices cgroup check as hard requirement

Commit:
Merge pull request #12796 from moxiegirl/add-to-docs-maintainers

Adding James and theJeztah to the list

Commit:
Merge pull request #12798 from dnephin/add_missing_docs_about_labels

Add missing API docs about filtering by label

thaJeztah committed:
Merge pull request #12817 from WeiZhang555/man

improve docker man page

hqhq committed:
fix a minor inspect format issue

Before, inspect cont1 cont2 shows:
[{
    xxx
}
,{
    xxx
}
]
After, it shows:
[
{
   xxx
}
,{
    xxx
}
]

Because `func (*Encoder) Encode` always followed by a newline character,
so it's difficult to put '}' and ']' one the same line.

To get symmetry, above is our choice.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

Commit:
Merge pull request #12723 from moxiegirl/carry-11296

Integrating Windows Testing Material

estesp committed:
Merge pull request #12826 from cpuguy83/12825_fix_leftover_container_json

Fix undead containers

LK4D4 committed:
Merge pull request #12844 from hqhq/hq_fix_inspect_format

fix a minor inspect format issue

Commit:
Merge pull request #12814 from vbatts/vbatts-rhel_updates

RHEL updates

Commit:
Merge pull request #12644 from psftw/official-repos-rewrite

Rewrite Official Repos page.

jfrazelle committed:
Merge pull request #12245 from buddhamagnet/dockerignore-ignores

Add support for exclusion rules in dockerignore

calavera committed:
Merge pull request #12820 from runcom/yet-another-sockRequest-refactor

Expose whole Response struct in sockRequestRaw

jfrazelle committed:
Merge pull request #12789 from LK4D4/daemon_suite

Implement daemon suite for integration-cli

cpuguy83 committed:
Merge pull request #11190 from cyphar/expose-path-sanitisation-wrappers

Expose the getResourcePath and getRootResourcePath wrappers.

coolljt0725 committed:
Fix docker rename help not consistent with other commands

Signed-off-by: Lei Jitang <leijitang@huawei.com>

cpuguy83 committed:
Merge pull request #12870 from coolljt0725/fix_rename_help

Fix docker rename help not consistent with other commands

vdemeester committed:
Add coverage on pkg/fileutils

Should fix #11598

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

duglin committed:
Merge pull request #12840 from vdemeester/11598-pkg-fileutils-tests

Add coverage on pkg/fileutils

jfrazelle committed:
Merge pull request #10427 from jfrazelle/option-to-disable-systemd-cgroups

Option to configure cgroup manager (adds --exec-opt)

LK4D4 committed:
Merge pull request #11526 from mizzy/append-drop-rule-not-insert

Append DROP rule with --icc=false, not insert

jfrazelle committed:
Merge pull request #12875 from runcom/remove-cfatal-in-goroutine-integration-cli

c.Fatal won't fail and exit test inside a goroutine

estesp committed:
Merge pull request #12439 from Mashimiao/add-err-check-before-getting-term

Add error check before getting term

thaJeztah committed:
Merge pull request #12574 from ahmetalpbalkan/docs/systemd-dir

docs: Add more places docker.service can be at

boucher committed:
Update checkpoint/restore to use the new server/daemon pattern

boucher committed:
Updates to docker master
